torch/_inductor/fx_passes/fuse_attention.py

```diff
@@ -141,6 +146,44 @@ def _sfdp_replacement_6(query, key, value, attn_mask, dropout_p):
 # TODO(jansel): make these pattern work with lowmem_dropout=True
 
 
+def _sfdp_scale_factor_check(scale_factor_op):
+    def fn(match):
+        # make sure the mul(div) for the scale factor is scalar mul(div).
+        # bmm->view->mul(div)
+        matmuls = filter_nodes(match.nodes, aten.bmm)
+        if len(matmuls) < 2:
+            return False
+        if (
+            len(matmuls[0].users) != 1
+            or list(matmuls[0].users.keys())[0].target != aten.view.default
+        ):
+            return False
+        view_node = list(matmuls[0].users.keys())[0]
+        if (
+            len(view_node.users) != 1
+            or list(view_node.users.keys())[0].target != scale_factor_op
+        ):
+            return False
+        scale_factor_node = list(view_node.users.keys())[0]
```

Comment:
Could you go directly here with `filter_nodes(match.nodes, scale_factor_op)`?

Reply:
Yes, changed.